### ISSUE TITLE
[$1000] iOS - Deepliink - Keyboard displayed upon splash screen if open new chat deeplink

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#24473]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___
**Issue found when executing PR** https://github.com/Expensify/App/pull/24126
## Action Performed: 
Precondition - Make sure that the app is completely closed (killed) and doesn't run in the background
1.  Tap on https://staging.new.expensify.com/new/chat
deeplink (e.g. in notes)

## Expected Result:
"New chat" tab and keyboard appear after splash screen is dismissed

## Actual Result:
Keyboard appears upon loading splash screen.
In Production the issue is reproducible when user interacts with staging deeplinks. If user taps on production deeplinks they will land in mWeb, not iOS app

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [x] iOS / native
- [ ] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.53.1

**Reproducible in staging?:** Yes

**Reproducible in production?:** Yes

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/78819774/b47d8bf5-f107-4a43-a559-c0e7ff0f23b0

**Expensify/Expensify Issue URL:**

**Issue reported by:** Applause - Internal Team

**Slack conversation:**

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~0100e6a84621057a16</li>
        <li>Upwork Job ID: 1691899029638578176</li>
        <li>Last Price Increase: 2023-08-23</li>
    </ul>
</details>

**Comment by isabelastisser** [comment]
I was assigned this when I was OOO. I will review this soon!

**Comment by jjcoffee** [comment]
@rayane-djouah I don't think your RCA is correct, if I set `shouldFocusOnSelectRow` to false, the text input is still focused (as you'd expect).

@tienifr Leaning towards your proposal, but I'd just like to understand the benefit of your main solution over the alternative, which sounds simpler to me at first glance! What is the benefit of using the `useFocusOnTransitionEnd` hook, over simply checking for `isSplashHidden` in `componentDidUpdate`?

Can either of you still replicate the issue, as these related issues are being [reported](https://github.com/Expensify/App/issues/22508#issuecomment-1679606222) as no longer occurring on staging or prod.

**Comment by jjcoffee** [comment]
@tienifr Thanks, that makes sense! I will come back for a full review on Monday.

**Comment by jjcoffee** [comment]
@tienifr Weird, I can't replicate on an iOS simulator (on latest main). Were you replicating on a physical device? Looks like there have been several changes to `BaseOptionsSelector` so that it no longer directly focuses in `componentDidMount` so maybe this is no longer an issue.

@lanitochka17 are you able to retest on latest main?

**Comment by jjcoffee** [comment]
@tienifr Haha yes I did have the keyboard turned on! 

I'm no longer sure if this is really a bug if it now always works the same as in your repro video. The keyboard is showing _just as_ the splash screen is transitioning, rather than way before like in the video in the issue description. I feel like that's acceptable behaviour. Asking for feedback [on Slack](https://expensify.slack.com/archives/C01GTK53T8Q/p1692711702275149).

Separately, your RCA doesn't cover why the issue would be happening on iOS but not on Android. Do you have any thoughts on that?

**Comment by isabelastisser** [comment]
@jjcoffee I bump[ your questio]([url](https://expensify.slack.com/archives/C01GTK53T8Q/p1692711702275149))n in Slack!

**Comment by jjcoffee** [comment]
@isabelastisser We got two votes for not a bug, not sure if that's enough to close? :smile:

**Comment by lanitochka17** [comment]
Issue reproducible on Build 1.3.56-18
Iphone 14 Pro Max/ios16.6

https://github.com/Expensify/App/assets/78819774/ac0c053a-4e26-447b-b7a6-78f250b0d088

**Comment by isabelastisser** [comment]
Closing this per slack discussion.

**Comment by jjcoffee** [comment]
@tienifr Ah yeah that wasn't clear to me before. Then it's more of an inconsistency issue, although I still think it's a fairly benign edge case unless there's an example of a really long delay between keyboard displaying and splash screen hiding.

Based on your [proposal](https://github.com/Expensify/App/issues/24473#issuecomment-1681538086)'s RCA, would you say this issue would happen on _any_ screen that uses `OptionsSelector` then? Maybe if we expand the issue to cover all cases where this would happen, it'll be more worth fixing!

**Comment by jjcoffee** [comment]
Yeah I'm leaning towards reopening, especially seeing @lanitochka17's video, it's a bit more obvious. Raising [on Slack](https://expensify.slack.com/archives/C01GTK53T8Q/p1693230295022269) for some more opinions.

**Comment by isabelastisser** [comment]
thanks @jjcoffee ! I bumped the discussion now.

**Comment by isabelastisser** [comment]
Confirmed that we should reopen this!

**Comment by isabelastisser** [comment]
Hey @jjcoffee, what are the next steps? Do we have a good proposal already, or should we wait for more?

**Comment by jjcoffee** [comment]
@isabelastisser Since the issue's scope has changed a bit to cover all pages which use `OptionsSelector`, I think we could do with a bit of extra detail in the RCA of @tienifr's proposal (or anyone else who wants to propose!). Specifically, I'm curious why the issue occurs only on any page that uses `OptionsSelector`, but not other pages like the workspace members invite page. It would also be good to understand why the issue occurs only on iOS.

@tienifr if you have a test branch to share that would also be great!

**Comment by jjcoffee** [comment]
@tienifr Sorry for the delay and thanks for the update, the RCA is much clearer now! If you have it already, a test branch of the main solution would be amazing (it's not a requirement, of course).

It's interesting that we're relying on timeouts (via `shouldDelayFocus`) on other screens as presumably a workaround for this issue. It makes sense to have a consistent solution for this issue, so I think it makes sense to either implement the same (`shouldDelayFocus`) here or implement a new (hopefully better) solution across all screens. I guess the question is if your solution handles Android animation transition issues equally well (which is I believe the other reason `shouldDelayFocus` is used).

**Comment by isabelastisser** [comment]
@jjcoffee, do we still need more proposals?

**Comment by jjcoffee** [comment]
@isabelastisser Still open for other proposals, just waiting on @tienifr's response to my [comment](https://github.com/Expensify/App/issues/24473#issuecomment-1705503926). I may need to open up for broader discussion on Slack, as we need to ideally align on a generic app-wide solution here, I think!

**Comment by jjcoffee** [comment]
@tienifr Thanks that's really helpful! Do you think we should hold for https://github.com/Expensify/App/issues/22850 where `onEntryTransitionEnd` doesn't fire? Looks like you're [close](https://github.com/Expensify/App/issues/22850#issuecomment-1708068160) to being assigned there, and it seems to make sense to fix that issue first as without it we might end up not focusing the input at all, which wouldn't be ideal!

**Comment by jjcoffee** [comment]
@isabelastisser Can you update the title to hold on https://github.com/Expensify/App/issues/22850? Thanks!

**Comment by isabelastisser** [comment]
Not overdue, still on hold!

**Comment by isabelastisser** [comment]
Thanks for clarifying, @tienifr ! I changed it to Daily, can you please confirm what's next here? Do you need to update your proposal?

cc @jjcoffee

**Comment by isabelastisser** [comment]
Bump @jjcoffee can you please review the updated proposal above? Thanks!

**Comment by jjcoffee** [comment]
Sorry higher priority PR got in the way, will review tomorrow!

**Comment by jjcoffee** [comment]
@tienifr I believe we may have settled on a standard pattern for this [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1694660990479979). Can you comment on whether or not you think that's applicable for this case?

**Comment by jjcoffee** [comment]
Great, thanks for clarifying! 

I'm happy to go with @tienifr's [proposal](https://github.com/Expensify/App/issues/24473#issuecomment-1681538086) in that case!

:ribbon::eyes::ribbon: C+ reviewed

**Comment by jjcoffee** [comment]
Just a quick update - PR has been approved by me, so just waiting for @danieldoglas to approve.

**Comment by isabelastisser** [comment]
Bump @danieldoglas for the comment above. Thanks!

**Comment by jjcoffee** [comment]
@isabelastisser PR has been merged!

**Comment by danieldoglas** [comment]
interesting, this was deployed to production but it didn't post the summary yet.

It was deployed on the 9th, so this should be good for payment next tomorrow @isabelastisser

**Comment by jjcoffee** [comment]
@isabelastisser Friendly bump for payment, checklist is completed [above](https://github.com/Expensify/App/issues/24473#issuecomment-1821032828)!

**Comment by danieldoglas** [comment]
@isabelastisser I think it should be created
